snap/naming/validate.go

```diff
@@ -174,3 +175,45 @@ func ValidateSnapID(id string) error {
 	}
 	return nil
 }
+
+var errInvalidSecurityTag = errors.New("invalid security tag")
+
+// ValidateSecurityTag validates known variants of snap security tag.
+//
+// Two forms are recognised, one for apps and one for hooks. Other forms
+// are possible but are not handled here.
+//
+// TODO: handle the weird udev variant.
+func ValidateSecurityTag(tag string) error {
+	parts := strings.SplitN(tag, ".", 4)
```

Comment:
The SplitN() change is fine but I actually preferred Split() since SplitN(..., 4) with something that would've had more parts forces the additional check in the ValidateApp()/ValidateHook() rather than erroring early off the simple length check. Not a blocker, just an observation.

Reply:
Indeed, I'll change it to SplitN(..., 5) so that the length check remains an early warning and add a comment as to why.